lib/spell-check-manager.coffee

```diff
@@ -0,0 +1,306 @@
+class SpellCheckerManager
+  checkers: []
+  locales: []
+  localePaths: []
+  useLocales: false
+  localeCheckers: null
+  knownWords: []
+  addKnownWords: false
+  knownWordsChecker: null
+
+  addPluginChecker: (checker) ->
+    console.log "spell-check: addPluginChecker:", checker
+    @addSpellChecker checker
+  addSpellChecker: (checker) ->
+    console.log "spell-check: addSpellChecker:", checker
+    @checkers.push checker
+
+  removeSpellChecker: (spellChecker) ->
+    @checkers = @checkers.filter (plugin) -> plugin isnt spellChecker
+
+  check: (args, text) ->
+    # Make sure our deferred initialization is done.
+    @init()
+
+    # We need a couple packages.
+    multirange = require 'multi-integer-range'
```

Comment:
I thought about a word-based API, but then I was thinking that the overhead of native interop would be more inefficient. At least in C# and Java, there is more overhead to jump to the native libraries (as opposed to a single large string once) and I thought a word-by-word would be excessive. But, if that isn't a problem, then the `multi-integer-range` wouldn't be needed because we could stop earlier.

My general rule was if any checker says it is correct, it is correct. We do that in lines [45-47](https://github.com/dmoonfire/spell-check/blob/master/lib/spell-check-manager.coffee#L45-L47) by a simple union of indexes. Known words and `spell-check-project` both only do correctness checking. I mainly union'd those together because it was more efficient than just inverting the range and then adding them into the second rule (below).

The second rule was a word was incorrect if every checker says it is incorrect. The multiple locale checkers use this rule. To do that, we keep a integer range of incorrect words for each checker then get the intersection of all of them to identify which words are incorrect in every single one. I mainly used `multi-integer-range` to make the intersection logic a bit easier to understand using set theory instead of managing it myself (I really hate writing set operations :P).

Doing word-by-word would prevent [#13](https://github.com/atom/spell-check/issues/13) but I think that is a grammar checker thing, not a spell-checking.